src/context/unconfirmed-transactions-repository.ts

```diff
@@ -46,6 +47,18 @@ export class UnconfirmedTransactionRepository extends BaseRepository<Unconfirmed
         filterQuery = filterQuery.setParameters({ ergoTree });
       }
 
+      if (options.where?.transactionId && transactionIds) {
+        transactionIds.push(options.where.transactionId);
+        delete options.where.transactionId;
+      }
+
+      if (transactionIds) {
```

Comment:
If an empty array is submitted, it will result in a broken query, please add a check for it.

Reply:
Fixed.